timemachine/potentials/nonbonded.py

```diff
@@ -1,24 +1,50 @@
+import jax
+jax.config.update("jax_enable_x64", True)
 import jax.numpy as np
 from jax.scipy.special import erfc
 from jax.ops import index_update, index
-from timemachine.potentials.jax_utils import distance, convert_to_4d
+from timemachine.potentials.jax_utils import distance_on_pairs, distance, convert_to_4d, get_all_pairs_indices
 
 
 def switch_fn(dij, cutoff):
-    return np.power(np.cos((np.pi*np.power(dij, 8))/(2*cutoff)), 2)
+    return np.power(np.cos((np.pi * np.power(dij, 8)) / (2 * cutoff)), 2)
+
+
+from typing import Optional
+
+Array = np.array
+
+
+def lennard_jones(dij, sig_ij, eps_ij):
+    """https://en.wikipedia.org/wiki/Lennard-Jones_potential"""
+    sig6 = (sig_ij / dij) ** 6
+    sig12 = sig6 ** 2
+
+    return 4 * eps_ij * (sig12 - sig6)
+
+
+def direct_space_pme(dij, qij, beta):
+    """Direct-space contribution from eq 2 of:
+        Darden, York, Pedersen, 1993, J. Chem. Phys.
+        "Particle mesh Ewald: An N log(N) method for Ewald sums in large systems"
+        https://aip.scitation.org/doi/abs/10.1063/1.470117
+    """
+    return qij * erfc(beta * dij) / dij
 
 
 def nonbonded_v3(
-    conf,
-    params,
-    box,
-    lamb,
-    charge_rescale_mask,
-    lj_rescale_mask,
-    beta,
-    cutoff,
-    lambda_plane_idxs,
-    lambda_offset_idxs):
+        conf,
+        params,
+        box,
+        lamb,
+        charge_rescale_mask,
+        lj_rescale_mask,
+        beta,
+        cutoff,
+        lambda_plane_idxs,
+        lambda_offset_idxs,
+        runtime_validate=False,
```

Comment:
When is this set to True? I noticed that most of the tests have this set to False

Reply:
Ahh, I changed the default behavior here. Previous behavior corresponds to runtime_validate=True , following https://github.com/proteneer/timemachine/pull/424#discussion_r629678467 . Will revert the default behavior to True

Reply:
Addressed in https://github.com/proteneer/timemachine/commit/840c9a09be58b075aab760ce1b71d0d3df33eb5c . Default behavior (True) now matches previous behavior, and non-default behavior (False) remains used in tests that require JIT.